docs/helpers.rst

```diff
@@ -191,6 +191,26 @@ transaction support.  This is only required for fixtures which need
 database access themselves.  A test function would normally use the
 :py:func:`~pytest.mark.django_db` mark to signal it needs the database.
 
+``shared_db_wrapper``
+~~~~~~~~~~~~~~~~~~~~~
+
+This fixture can be used to create long-lived state in the database.
+It's meant to be used from fixtures with scope bigger than ``function``.
+It provides a context manager that will create a new database savepoint for you,
+and will take care to revert it when your fixture gets cleaned up.
+
+At the moment it does not work with ``transactional_db``,
+as the fixture itself depends on transactions.
+It also needs Django >= 1.6, as it's based on the ``atomic`` API.
+
+Example usage::
+
+  @pytest.fixture(scope='module')
+  def some_users(request, shared_db_wrapper):
+      with shared_db_wrapper(request):
```

Comment:
I would like that too, but there is an issue with this. When you specify a dependency to `some_users`, the `shared_db_wrapper` fixture will still be active. This is a problem, because you only want to use global state for a limited amount of data (e.g. creating the users and not modifying them).

I think that is why OP has called it a `shared_db_wrapper` and not `shared_db`. 

However, maybe there's a solution around this. Maybe it's possible to somehow create a special "destructor" before entering "local" scope. However this might be complicated given how the current architecture (I have no idea about pytest-django internals).